test/extended/cli/builds.go

```diff
@@ -0,0 +1,339 @@
+		args = append(args, fmt.Sprintf("%v-%v", bcName, buildNumber))
+	} else {
+		args = append(args, "-l", fmt.Sprintf("openshift.io/build-config.name=%v", bcName))
+	}
+
+	return wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
+		out, err := oc.Run("get").Args(args...).Output()
+
+		if err != nil {
+			return false, nil
+		}
+		lines := strings.Split(out, "\n")
+
+		return len(lines) >= 2, nil
+	})
+}
+
+func getExpectedBCOutputMatcher(bcName string, isNames ...string) t.GomegaMatcher {
+	matcher := o.ContainSubstring("Success")
+	if bcName != "" {
+		matcher = o.And(matcher, o.ContainSubstring(fmt.Sprintf("buildconfig.build.openshift.io \"%v\" created", bcName)))
+	} else {
+		e2e.Logf("no bcName specified")
+	}
+
+	for _, isName := range isNames {
+		if isName != "" {
+			matcher = o.And(matcher, o.ContainSubstring(fmt.Sprintf("imagestream.image.openshift.io \"%v\" created", isName)))
+		}
+	}
+	return matcher
+}
+
+func getBCInfoByTemplate(oc *exutil.CLI, bcName, template string) string {
+	if bcName == "" {
+		e2e.Logf("no bcName specified")
+		return ""
+	}
+	out, err := oc.Run("get").Args("bc/"+bcName, "--template", template).Output()
+
+	if err != nil {
+		e2e.Logf("failed to get %v: %v", bcName, err)
+		return ""
+	}
+	return out
+}
+
+func getBCSourceType(oc *exutil.CLI, bcName string) string {
+	return getBCInfoByTemplate(oc, bcName, bcSourceTypeTemplate)
+}
+
+func getBCOutputType(oc *exutil.CLI, bcName string) string {
+	return getBCInfoByTemplate(oc, bcName, bcOutputTemplate)
+}
+
+var _ = g.Describe("[sig-cli] oc builds", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLI("oc-builds").AsAdmin()
+
+	g.Describe("new-build", func() {
+		g.It("build from a binary with no inputs requires name", func() {
+			out, err := oc.Run("new-build").Args("--binary").Output()
+			o.Expect(err).To(o.HaveOccurred())
+			o.Expect(out).To(o.ContainSubstring("you must provide a --name"))
+		})
+
+		g.It("build from a binary with inputs creates a binary build", func() {
+			out, err := oc.Run("new-build").Args("--binary", "--name=binary-test").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(out).To(getExpectedBCOutputMatcher("binary-test", "binary-test"))
+			o.Expect(getBCSourceType(oc, "binary-test")).To(o.Equal("Binary"))
+			o.Expect(getBCOutputType(oc, "binary-test")).To(o.Equal("ImageStreamTag binary-test:latest"))
+		})
+
+		g.It("build from git with output to ImageStreamTag", func() {
+			out, err := oc.Run("new-build").Args("centos/ruby-26-centos7", "https://github.com/openshift/ruby-hello-world.git").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(out).To(getExpectedBCOutputMatcher("ruby-hello-world", "ruby-hello-world"))
+			o.Expect(getBCSourceType(oc, "ruby-hello-world")).To(o.Equal("Git"))
+			o.Expect(getBCOutputType(oc, "ruby-hello-world")).To(o.Equal("ImageStreamTag ruby-hello-world:latest"))
+			o.Expect(waitForBuild(oc, "ruby-hello-world", "")).NotTo(o.HaveOccurred())
```

Comment:
I'm looking at the original test in https://github.com/openshift/origin/blob/d704a4d2ab5e55731d11770c11eacd666940b944/test/extended/testdata/cmd/test/cmd/builds.sh#L27-L29 and it doesn't wait for the build to succeed, actually, it only checks the definition of the newly created BuildConfig matches the expectation. Waiting for every build to complete unnecessarily lengthens the test time, which is not the goal. The build correctness is verified elsewhere, we care about oc here, mostly.

Reply:
I added one wait to the `start-build` test and removed the rest